internal/querynodev2/delegator/delegator_data.go

```diff
@@ -371,76 +371,78 @@ func (sd *shardDelegator) loadStreamDelete(ctx context.Context,
 	deltaPositions []*msgpb.MsgPosition,
 	targetNodeID int64, worker cluster.Worker) error {
 	log := sd.getLogger(ctx)
-	sd.deleteMut.Lock()
-	defer sd.deleteMut.Unlock()
 
 	idCandidates := lo.SliceToMap(candidates, func(candidate *pkoracle.BloomFilterSet) (int64, *pkoracle.BloomFilterSet) {
 		return candidate.ID(), candidate
 	})
 
-	// apply buffered delete for new segments
-	// no goroutines here since qnv2 has no load merging logic
-	for i, info := range infos {
-		candidate := idCandidates[info.GetSegmentID()]
-		position := info.GetDeltaPosition()
-		if position == nil { // for compatibility of rolling upgrade from 2.2.x to 2.3
-			position = deltaPositions[i]
-		}
-
-		deleteData := &storage.DeleteData{}
-		// start position is dml position for segment
-		// if this position is before deleteBuffer's safe ts, it means some delete shall be read from msgstream
-		if position.GetTimestamp() < sd.deleteBuffer.SafeTs() {
-			log.Info("load delete from stream...")
-			var err error
-			deleteData, err = sd.readDeleteFromMsgstream(ctx, position, sd.deleteBuffer.SafeTs(), candidate)
-			if err != nil {
-				log.Warn("failed to read delete data from msgstream", zap.Error(err))
-				return err
+	{
```

Comment:
This block is equailant to whole block until the end of this function. Maybe it's not needed.